src/metabase/models/model_index.clj

```diff
@@ -70,10 +74,18 @@ don't, (and shouldn't) care that those are expressions. They are just another fi
                     {:field-ref field-ref
                      :valid-clauses [:field :expression]}))))
 
-(defn- fetch-values
-  [model-index]
+(mr/def ::model-index
```

Comment:
Tests in this namespace were flaky and stuff so while trying to debug what the hecc was wrong with it I worked backwards from existing tests and added some schemas so I could fail faster and find the source of problems. Leaving them in because they make lives easier